atlasdb-impl-shared/src/main/java/com/palantir/atlasdb/transaction/impl/SerializableTransactionManager.java

```diff
@@ -69,6 +66,31 @@ public LockService getLockService() {
         public void registerClosingCallback(Runnable closingCallback) {
             manager.registerClosingCallback(closingCallback);
         }
+
+        @Override
+        public boolean isInitialized() {
+            if (!isInitialized) {
+                tryInitialize();
+            }
+
+            return isInitialized;
+        }
+
+        private synchronized void tryInitialize() {
+            if (isInitialized || !tryInitializeLimiter.tryAcquire()) {
+                return;
+            }
+
+            try {
+                // SerializableTransactionManager should throw until the underlying KVS is initialized.
+                // TODO(ssouza): replace with KVS healthcheck status when that gets implemented.
+                manager.getKeyValueService().getClusterAvailabilityStatus();
+                isInitialized = true;
+            } catch (NotInitializedException e) {
+                log.info("The KeyValueService is not initialized yet!");
```

Comment:
Do we even want to log here when a healthcheck fails? I'd only log in the delegate when `isInitialized()` returns false.